tcclient/http.go

```diff
@@ -69,12 +82,9 @@ func (connectionData *ConnectionData) APICall(payload interface{}, method, route
 		if reflect.ValueOf(payload).IsValid() && !reflect.ValueOf(payload).IsNil() {
 			ioReader = bytes.NewReader(jsonPayload)
 		}
-		u, err := url.Parse(connectionData.BaseURL + route)
+		u, err := setURL(connectionData, route, query)
 		if err != nil {
-			return nil, nil, fmt.Errorf("apiCall url cannot be parsed: '%v', is your BaseURL (%v) set correctly?\n%v\n", connectionData.BaseURL+route, connectionData.BaseURL, err)
-		}
-		if query != nil {
-			u.RawQuery = query.Encode()
+			return nil, nil, fmt.Errorf("apiCall url cannot be parsed:\n%v\n", err)
```

Comment:
We should probably agree some standards for errors, as you are right, there is not a clear policy. I'm often wrapping errors to give a clearer chain (in cases where it is thrown down several steps in the stack, it leaves an audit trail). But yes - it is something I've been thinking about formalising - so thanks for reminding me.